libs/pilight/core/datetime.c

```diff
@@ -612,22 +612,26 @@ time_t datetime2ts(int year, int month, int day, int hour, int minutes, int seco
 int tzoffset(char *tz1, char *tz2) {
 	logprintf(LOG_STACK, "%s(...)", __FUNCTION__);
 
+	if (strcmp(tz1, tz2) == 0) {	// prevent heavy calculations on simple case
+		return 0;
+	}
+
 	atomiclock();
 
 	time_t utc, tzsearch, now;
 	struct tm tm;
-	char oritz[64];
+	char oritz[64], s;
```

Comment:
it is wrong to save pointers of getenv() return values for alonger time (in particular if the environment will be modified later on, as is the case here) because the saved pointer became invalid after environment modifications. If you need the value later, you must store it in an own buffer (as oritz). There is still a bug if the TZ value in the environment is longer than be stored in oritz (64 chars).This could be fixed by allocating a buffer e.g. useing strdup.

It is ok to save the return value of getenv() to prevent an immediately followin getenv() call for the same variable. This is, what I did. Better is to set s=NULL after usage to prevent accidental use of s. I will chage that.